### PR TITLE
Minor Cleanup to Project and Hardhat Configuration

### PR DESCRIPTION
**Pull request by nlordell**
In preparation for #145, I was making sure that the deployments work on Geth and found some small cleanups that I would like to do in a separate PR (just to reduce the noise from any followups).

Namely:
- Move safe contract dependency imports to their own file instead of `SafeMock.sol`.
- Use single solidity version (we had 3 before, but only 0.8.21 was actually ever used to build anything)
- Use SafeL2 in the E2E test for two reasons:
  1. it does slightly more things than `Safe` which is worth testing
  2. we explicitly depend on `SafeL2` in the new `SafeContracts.sol` file, while `Safe` gets pulled in “by chance”.
- Bump Solidity to the latest 0.8.23 version

### CODE REVIEW

File: 4337/hardhat.config.ts [hunk 1]
```diff
@@ -25,16 +25,33 @@ if (PK) {
...
   throw new Error(`Could not find Infura key in env, unable to connect to network ${argv.network}`)
 }
 
 import './src/tasks/local_verify'
 import './src/tasks/deploy_contracts'
 import './src/tasks/show_codesize'
 
-const primarySolidityVersion = SOLIDITY_VERSION || '0.8.21'
-const soliditySettings = !!SOLIDITY_SETTINGS ? JSON.parse(SOLIDITY_SETTINGS) : { optimizer: { enabled: true, runs: 10_000_000 } }
+const solidityVersion = SOLIDITY_VERSION || '0.8.21'
```
Comment (mmv08) [inline 4337/hardhat.config.ts:36]:
```suggestion
const solidityVersion = SOLIDITY_VERSION || '0.8.23'
```

Comment (mmv08) [reply 4337/hardhat.config.ts:36]: I think we can also pump the solc version :)

**Review by mmv08 — APPROVED** [review]

File: 4337/hardhat.config.ts (hunk 1 above)
Comment (nlordell) [reply 4337/hardhat.config.ts:36]: I would keep that change separate for a couple reasons:
- It means that we may potentially have the Safe4337Module end up on a different address that what is currently in the deployments
- This PR is intended to not change behaviour at all

Comment (nlordell) [reply 4337/hardhat.config.ts:36]: **Edit**: I think you are right… we are already developing for a 0.2 version anyway, so bumping makes sense.

Comment (mmv08) [reply 4337/hardhat.config.ts]: > I would keep that change separate for a couple reasons:

> It means that we may potentially have the Safe4337Module end up on a different address that what is currently in the deployments

Whoever wants to reproduce the deployment should use the release tag anyway :)

**Pull request merged**